Dev omnikit podinforesponse

Lytrix opened the pull request:
New PR, did a merge on the dev branch as a new branch.
First tried rebasing/merging the dev on my branch but that was messy. This will ensure the project and plist whitespaces remain untouched.

Lytrix committed:
* Created StatusMessageBlock to switch to proper StatusError Type Decoding struct
* Created StatusResponseConfiguredAlerts
* Created StatusReponseFaultEvent

Lytrix committed:
* Added StatusResponseConfiguredAlerts, StatusResponseFaultEvent and StatusMessageBlock to Xcode Project
* Added BolusTests and StatusTests to Xcode Project

Lytrix committed:
Made StatusTests working again with new StatusMessageBlock

Lytrix committed:
Started on creating StatusResponseConfiguredAlerts tests

Lytrix committed:
Removed StatusReponseFaultEvent
Pointing Type encodedData[2] to StatusMessageBlock

Lytrix committed:
Made StatusErrorConfiguredAlerts working
Ealuates every alertType 1-8, can return an array of settings with beeptype

Lytrix committed:
Renamed pulsesLeft > unitsLeft

Lytrix committed:
Renamed statuserror test to explain them better

Lytrix committed:
Added more test:
* Replace Pod after 3 days limit
* Replace Pod after Reservoir is depleted (forcing to deactivate it)

Lytrix committed:
Added 0x11 AcknowledgeAlertCommand with tests

Lytrix committed:
Renamed StatusError to PodInfo

Lytrix committed:
Renamed PodInfoMessageBlock > PodInfoBlock

Lytrix committed:
Renamed  PodInfo -> PodInfoResponse, PodInfoBlock -> PodInfo

Lytrix committed:
* Renamed PodInfoType to PodResponseSubType
* Made the PodInfoResponse select the appropriate PodInfoResponse Type swift files.
PodInfoResponse.swift creates a PodInfoResponseSubType enum instance from encodedData[2], and call podInfoType on it to return the appropriate PodInfo type via init(), which can then be used to decode the rest of the data.

Lytrix committed:
Merge branch 'omnikit-podinforesponse-rebase-dev' into dev-omnikit-podinforesponse

Lytrix committed:
Fixed merge issues messageblock and project

Lytrix committed:
statusType -> podInfoResponseSubType

Lytrix committed:
reordered acknowledgeAlert

Lytrix committed:
Added PodInfoResetStatus 0x46 + tests

Lytrix committed:
Added tests for requesting 0x01 0x02 0x46 PodInfoSubTypes

Lytrix committed:
Added project updates to new files

Lytrix committed:
fixed wrong byte selection address

Lytrix committed:
Added resetstatus case to PodInfo options

Lytrix committed:
* Added current known FaultEventTypes to PodInfo.
* Added new PodInfo Type Datalog
* Updated PodInfoTests for new FaultEventTypes

Lytrix committed:
* Added FaultEventType into PodInfoResponseSubType block to find it back if used on other swift files.
* Updated other files accordingly.
* Fixed proper order of elements in PodInfoFaultEvent

Lytrix committed:
Renamed problemCVerifyStartPump > faultEventSetupPod

Lytrix committed:
added more decoded elements to Type3 Podinfo

Lytrix committed:
removed logdata value for now to pass test

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -47,64 +47,58 @@ public struct StatusError : MessageBlock {
             throw MessageBlockError.notEnoughData
         }
         
         self.length = encodedData[1]
```
Because this object is now being constructed with `podInfo = try podInfoResponseSubType.podInfoType.init(encodedData: encodedData.subdata(in: 2..<len-2))`, all the array accesses should be shifted minus 2.

Lytrix replied:
Fixed, as well as the tests for these types.

ps2 reviewed (changes requested):
I think we're close.  Looking forward to getting some of this fault information displayed in the UI

Lytrix committed:
Added PodInfoFault Type5 with tests

Lytrix committed:
Renamed faultEvent>faultEventType

Lytrix committed:
Fixed PodInfo encodedData input length and tests:
Omitted first 2 bytes, those are not passed through to the Podinfo decoding types.

Lytrix commented:
For the faultEventTypes I tried to make descriptive names, but I can also imagine a simpler case description like faultError92 would also suffice. What do you think is best? I do see some clustering of SetupPod errors and TempBasal errors which could be nice to see back in the code if errors are occuring, so left that way for now:
https://github.com/ps2/rileylink_ios/blob/145dbf7f84263d01606c0b4b8da88cc30e57bdc3/OmniKit/MessageTransport/MessageBlocks/PodInfo.swift#L46

Lytrix committed:
Moved DateComponents into PodInfoFault

Lytrix committed:
Added PodInfoFault to project

Lytrix committed:
Added PodInfoType 6 hardcodedTestValues

Lytrix committed:
Added PodInfoFlashLogRecent Type 50

Lytrix committed:
Merge remote-tracking branch 'upstream/dev' into dev-omnikit-podinforesponse

Lytrix committed:
Added PodInfoFlashLogRecent to project

Lytrix committed:
* Renamed resetStatus>flashVariables
* Added Type 51 to PodInfo
* Cleanup code

Lytrix commented:
Moved latest additions into PR on omnipod-testing.
